### DISCUSSION TITLE
Missing Data After Crash

### DISCUSSION THREAD

**Discussion by ChuckBland** [jasonacox/Powerwall-Dashboard#499] [General]
System: ACER Chromebook now running Debian, Powerwall 3

Checked on the computer this morning a little before 8am and found the computer locked-up around 4am. I restarted everything.

The attached photo shows the hole in the data between crash time and restart. I am presuming that the dashboard program can get the powerwall data on the Tesla server.

I did fiddle with start and stop times for the graph, etc, but it just won't fetch/display that data,. The data is there, as it shows on the Tesla app and Powerwall Companion.

Is there a way to make it retrieve that data?

Chuck

![20240704_172145](https://github.com/jasonacox/Powerwall-Dashboard/assets/11139777/8604f3e3-14e0-4c32-9fe7-9869cf041b5f)

**Comment by mcbirse** [comment]
Hi @ChuckBland - have you tried using the [Tesla-History](https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/tesla-history) import tool to retrieve the missing data?

You can also use the tool to retrieve all your historical data from when you had your Powerwall installed.

Also, from your graph, it appears you are getting weather data as well... as it happens I wrote a [Weather-History](https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/weather-history) import tool for that too. All bases covered. 😃 

Let us know how you go!
